massenergize/frontend-admin
# Admin logged out notification

Admin portal will now redirect automatically when admin is logged out after the app is left dormant for a long period of time. 

The idea is that the app sends a request to the backend every **30 minutes** to check if they are still signed in.

If they are not, all active tabs just get redirected back to the login page. 


**HOW TO TEST** 
1. Go local on the admin portal
2. Reduce the time it takes to run the request from 30minutes to say 5 seconds in the codebase. 
3. Sign in as an admin.
4. Now, go local in the api as well
5. Edit `middleware.py` function `_get_decoded_token` like this
```python 
 def _get_decoded_token(self, token) -> Tuple[dict, MassEnergizeAPIError]:
        return None, CustomMassenergizeError('session_expired')
```
6. Now save and come back to the admin portal to see the redirect in action.
